code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -120,32 +130,37 @@ func (msc *MinerSmartContract) GetGroupShareOrSignsHandler(ctx context.Context,
 	gsos := block.NewGroupSharesOrSigns()
 	groupBytes, err := balances.GetTrieNode(GroupShareOrSignsKey)
 	if err != nil {
-		return nil, err
+		err := smartcontract.NewError(smartcontract.FailRetrievingGroupErr, err)
+		return nil, smartcontract.WrapErrNoResource(err)
 	}
 	err = gsos.Decode(groupBytes.Encode())
 	if err != nil {
-		return "", err
+		err := smartcontract.NewError(smartcontract.FailDecodingGroupErr, err)
```

Comment:
Error could be more granular. Something like "can't decode group shares or signs".

Reply:
Done.